code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -39,6 +39,14 @@
 		REMOVE_TRAIT(user, TRAIT_MESON_VISION, "meson_glasses[UID()]")
 		active_on_equip = FALSE
 
+	if(mode == MODE_RAD)
+		if(!HAS_TRAIT_FROM(user, SM_HALLUCINATION_IMMUNE, "meson_glasses[UID()]"))
+			ADD_TRAIT(user, SM_HALLUCINATION_IMMUNE, "meson_glasses[UID()]")
+		active_on_equip = TRUE
+	else
+		REMOVE_TRAIT(user, SM_HALLUCINATION_IMMUNE, "meson_glasses[UID()]")
+		active_on_equip = FALSE
```

Comment:
This is gonna need a new solution, since theres 2 valid modes for it now. You'll need to refactor this to not use `active_on_equip` because it will always be false here.